craft_application/services/lifecycle.py

```diff
@@ -277,11 +282,23 @@ def get_primed_stage_packages(self, *, part_name: str) -> list[str] | None:
         """
         return self._lcm.get_primed_stage_packages(part_name=part_name)
 
-    def run(self, step_name: str | None, part_names: list[str] | None = None) -> None:
-        """Run the lifecycle manager for the parts."""
+    def run(
+        self,
+        step_name: str | None,
+        part_names: list[str] | None = None,
+        *,
+        build_info: craft_platforms.BuildInfo | None = None,
```

Comment:
I'm having a hard time finding the code that is now calling LifecycleService.run(build_info) that I imagine triggered this change; where is it?

Reply:
It was only inside the destructive mode form, so I've removed it. I've kept the expanded docstring though (but adjusted for not taking build_info)